internal/plugin/service/kafka/kafka_resource.go

```diff
@@ -0,0 +1,153 @@
+	return &kafkaResource{}
+}
+
+type kafkaResource struct {
+	client *aiven.Client
+	crud   service.Crud
+}
+
+func (s *kafkaResource) Configure(_ context.Context, req resource.ConfigureRequest, _ *resource.ConfigureResponse) {
+	if req.ProviderData == nil {
+		return
+	}
+
+	s.client = req.ProviderData.(*aiven.Client)
+	s.crud = service.NewCrud(s.client, service.TypeKafka)
+}
+
+func (s *kafkaResource) Metadata(_ context.Context, _ resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = "aiven_kafka"
+}
+
+func (s *kafkaResource) Schema(ctx context.Context, _ resource.SchemaRequest, resp *resource.SchemaResponse) {
+	resp.Schema = service.WithServiceCommon(ctx, schema.Schema{
+		Description: "The Kafka resource allows the creation and management of Aiven Kafka services.",
+		Attributes: map[string]schema.Attribute{
+			"default_acl": &schema.BoolAttribute{
+				Description: "Create default wildcard Kafka ACL",
+				Optional:    true,
+				Computed:    true,
+				PlanModifiers: []planmodifier.Bool{
+					boolplanmodifier.RequiresReplace(),
+				},
+				Default: booldefault.StaticBool(true),
+				//DiffSuppressFunc: schemautil.CreateOnlyDiffSuppressFunc,
+			},
+		},
+		Blocks: map[string]schema.Block{
+			"kafka": schema.SetNestedBlock{
+				Description: "Kafka server provided values",
+				NestedObject: schema.NestedBlockObject{
+					Attributes: map[string]schema.Attribute{
+						"access_cert": schema.StringAttribute{
+							Description: "The Kafka client certificate",
+							Computed:    true,
+							Sensitive:   true,
+						},
+						"access_key": schema.StringAttribute{
+							Description: "The Kafka client certificate key",
+							Computed:    true,
+							Sensitive:   true,
+						},
+						"connect_uri": schema.StringAttribute{
+							Description: "The Kafka Connect URI, if any",
+							Computed:    true,
+							Sensitive:   true,
+						},
+						"rest_uri": schema.StringAttribute{
+							Description: "The Kafka REST URI, if any",
+							Computed:    true,
+							Sensitive:   true,
+						},
+						"schema_registry_uri": schema.StringAttribute{
+							Description: "The Schema Registry URI, if any",
+							Computed:    true,
+							Sensitive:   true,
+						},
+					},
+				},
+			},
+			"kafka_user_config": kafka.NewResourceSchema(),
+		},
+	})
+}
+
+func (s *kafkaResource) Create(ctx context.Context, req resource.CreateRequest, resp *resource.CreateResponse) {
+	o := new(kafkaModel)
+	s.crud.Create(ctx, req, resp, o)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	if !o.DefaultACL.ValueBool() {
+		err := s.deleteDefaultACL(ctx, o.Project.ValueString(), o.ServiceName.ValueString())
+		if err != nil {
+			resp.Diagnostics.AddError("plan copy error", err.Error())
+			return
+		}
+	}
+}
+
+func (s *kafkaResource) Read(ctx context.Context, req resource.ReadRequest, resp *resource.ReadResponse) {
+	o := new(kafkaModel)
+	s.crud.Read(ctx, req, resp, o)
+}
+
+// Update todo
```

Comment:
nit: remove `todo` comment or add more context if there is something specific that needs to be done